### PR TITLE
Added mate HW5.1 Solution

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,74 @@
...
+        this.doors = doors;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
+        Car car = (Car) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: should we use super.clone()?

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,68 @@
...
+        this.cylinders = cylinders;
+    }
+
+    public void setCylindersPosition(String cylindersPosition) {
+        this.cylindersPosition = cylindersPosition;
+    }
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
+        return (Engine) super.clone();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Engine.java]: same

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,74 @@
...
 
-public class Car {
+public class Car implements Cloneable {
+    private String model;
+    private String colour;
+    private int seats;
+    private int doors;
+    private Engine engine;
 
+    public Car(
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Car.java]: this constructors are to long :)

Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Car.java]: Should I use builder here?

Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Car.java]: Shortened it to 1 line

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Engine.java]: Fixed.

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (AlexanderSashchenko) [reply src/main/java/core/basesyntax/Car.java]: Fixed.

**Pull request closed without merging**